Add an allowlist for api-test suites that can be ran in parallel

Smackteo opened the pull request:
#### b1ac99a11869889764a664c8bab40532c8d037c5
<pre>
Add an allowlist for api-test suites that can be ran in parallel
<a href="https://bugs.webkit.org/show_bug.cgi?id=299178">https://bugs.webkit.org/show_bug.cgi?id=299178</a>
<a href="https://rdar.apple.com/160937798">rdar://160937798</a>

Reviewed by Jonathan Bedard.

Adding an allowlist with suites that have passed local testing. Also changing DarwinPort to use it. Running the system shard last for stability.

* Tools/Scripts/webkitpy/api_tests/allowlist.txt: Added.
* Tools/Scripts/webkitpy/port/darwin.py:
(DarwinPort._load_api_test_suite_allowlist):
(DarwinPort):
(DarwinPort._should_use_system_shard):
(DarwinPort.sharding_groups):
*Tools/Scripts/webkitpy/api_tests/runner.py:

Canonical link: <a href="https://commits.webkit.org/300539@main">https://commits.webkit.org/300539@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/43fb3eeb4386182b6c831332cf63a9f104308b7c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/122980 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/42694 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/33391 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/129634 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/75089 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/a9f2585c-b6fe-4b07-b362-6e56ca702e2a) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/124857 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/43417 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/51288 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/93486 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/75089 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/4084b128-261b-43b8-884e-8d22e36c3cc6) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/125931 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/34611 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/110082 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/74113 "Passed tests") | | ⏳ 🛠 vision-apple 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/122336 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/33589 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/28235 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/73157 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/104321 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/28450 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/132363 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/49929 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/38017 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/101986 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/50306 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/106291 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/101855 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25873 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47218 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/25413 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/46703 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/49785 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/49252 "Built successfully") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/52605 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/50934 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e7db1c7bf0ba660913ebcb3ff66690cc30ef12eb. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51008

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/726193bf73ccc6413e81aa599e04b550af77b21b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51008

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/725f4f507b8ec93c81fa31ae32f070fd457de65d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51008

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5eaa6f5c83a7ca765ad85eb382b52cd29190f5ba. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51008

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/8072ebaec82e67e71678971f87933964d2fa6178. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51008

JonWBedard commented on Tools/Scripts/webkitpy/port/darwin.py:
```diff
@@ -53,10 +53,32 @@ def __init__(self, host, port_name, **kwargs):
+        if not self._filesystem.exists(allowlist_path):
+            return set()
+
+        allowlist = set()
```
Using a set is almost always correct, but I think in this case, you want to either use a traditional list or sort the set before returning it as a list. Python does not predictably order sets, and as you have learned, API tests are sometimes inappropriately dependent on ordering.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/80a1b6c0534bb4fcdae3d8459e93dc2a7711a037. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51008

Smackteo committed:
Add an allowlist for api-test suites that can be ran in parallel
https://bugs.webkit.org/show_bug.cgi?id=299178
rdar://160937798

Reviewed by Jonathan Bedard.

Adding an allowlist with suites that have passed local testing. Also changing DarwinPort to use it. Running the system shard last for stability.

* Tools/Scripts/webkitpy/api_tests/allowlist.txt: Added.
* Tools/Scripts/webkitpy/port/darwin.py:
(DarwinPort._load_api_test_suite_allowlist):
(DarwinPort):
(DarwinPort._should_use_system_shard):
(DarwinPort.sharding_groups):
*Tools/Scripts/webkitpy/api_tests/runner.py:

Canonical link: https://commits.webkit.org/300539@main
